File: ceph-asterisk-front/frontend/src/types/cdr.ts
export interface CDRRecord {
  id: number
  answer: string
  end: string
  clid: string
  src: string
  dst: string
  duration: number
  billsec: number
  disposition: string
  uniqueid: string
  userfield: string
  instance_name: string
  accountcode?: string
  dcontext?: string
  channel?: string
  dstchannel?: string
  lastapp?: string
  lastdata?: string
  amaflags?: number
}

export interface CDRFilter {
  instance_name?: string
  src?: string
  dst?: string
  date_from?: string
  date_to?: string
  disposition?: string
  limit?: number
  offset?: number
}

export interface CallRecord {
  answerDateTime: string
  endDateTime: string
  from: string
  to: string
  duration: string
  status: string
  vats: string
}

export interface CDRResponse {
  total: number
  items: CDRRecord[]
  limit: number
  offset: number
}

export interface CDRQueryParams {
  limit: number
  offset: number
  src?: string
  dst?: string
  disposition?: string
  date_from?: string
  date_to?: string
  instance_name?: string
}

File: ceph-asterisk/app/schemas/cdr.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Optional


class CDRRecord(BaseModel):

    dst: str
    start: Optional[datetime]=None
    userfield: Optional[str]=None
    dcontext: str
    answer: Optional[datetime]=None
    sequence: int
    clid: str
    end: Optional[datetime]=None
    channel: str
    duration: int
    dstchannel: Optional[str]=None
    billsec: int
    lastapp: str
    disposition: str
    src: str
    lastdata: Optional[str]=None
    amaflags: int
    accountcode: Optional[int]=None
    uniqueid: str #name-id
    instance_name:str
    
    model_config = ConfigDict(from_attributes=True)

class CDRRecords(BaseModel):
    total: int
    items: list[CDRRecord]
    limit: int
    offset: int
    

class CDRInputData(BaseModel):
    instance_name: Optional[str] = None
    src: Optional[str] = None
    dst: Optional[str] = None
    date_from: Optional[str] = None
    date_to: Optional[str] = None
    limit: int = 100
    offset: int = 0
    disposition: Optional[str]=None